lib/Dialect/TritonGPU/Transforms/ReorderInstructions.cpp

```diff
@@ -45,42 +45,36 @@ class TritonGPUReorderInstructionsPass
   void runOnOperation() override {
     ModuleOp m = getOperation();
     mlir::DominanceInfo dom(m);
-    // Sink conversions into loops when they will increase
-    // register pressure
-    DenseMap<Operation *, Operation *> opToMove;
     auto moveAfter = [](Operation *lhs, Operation *rhs) {
       auto lhsId = getWSRoleId(lhs);
       auto rhsId = getWSRoleId(rhs);
       if (lhsId == rhsId)
         lhs->moveAfter(rhs);
     };
+    // Move convert immediately after oprand ready
+    m.walk([&](triton::gpu::ConvertLayoutOp op) {
+      Operation *argOp = op.getOperand().getDefiningOp();
+      if (!argOp)
+        return;
+      moveAfter(op, argOp);
+    });
+    // reduce register pressure
+    // delay loading from shared memory to register,
+    // but never crossing loop
+    DenseMap<Operation *, Operation *> opToMove;
     m.walk([&](triton::gpu::ConvertLayoutOp op) {
       if (!willIncreaseRegisterPressure(op))
         return;
       auto user_begin = op->user_begin();
-      auto user_end = op->user_end();
-      if (std::distance(user_begin, user_end) != 1)
-        return;
-      if (user_begin->getParentOfType<scf::ForOp>() ==
+      if (user_begin->getParentOfType<scf::ForOp>() !=
```

Comment:
OK, here you flipped the sinking condition

Reply:
what about cases where you do need to reduce register pressure?

Reply:
Will, that is a trade off. Currently I don't have an idea about how to achieve that. Make compiler aware of shared memory and register usage on ttfir level is quite difficult I think.